[logs] document beforeSend context edition (#1000)
DataDog/browser-sdk: packages/logs/README.md

The `beforeSend` callback function gives you access to each event collected by the Browser SDK before it is sent to Datadog, and lets you update commonly redacted properties.

> Find examples of `beforeSend` use-cases in [the dedicated guide][5]. For example, to redact email addresses from your web application URLs:

Reviewer: ```suggestion
For examples of`beforeSend`, see [Enrich and control browser RUM data with beforeSend][5]. 

To redact email addresses from your web application URLs:
```

The `beforeSend` callback function gives you access to each event collected by the Browser SDK before it is sent to Datadog, and lets you update commonly redacted properties.

> For examples of`beforeSend`, see [Enrich and control browser RUM data with beforeSend][5]. 

Reviewer 2: ```suggestion
For examples of `beforeSend`, see [Enrich and control browser RUM data with beforeSend][5]. 
```

> To redact email addresses from your web application URLs:

Reviewer 2: Missing link here

Author: Why?

Reviewer 2: Ah, my bad, I got confused by the links modification below. I'll add a comment there.

| `message`       | String | The content of the log.                                                                          |
| `error.stack`   | String | The stack trace or complementary information about the error.                                    |
| `http.url`      | String | The HTTP URL.                                                                                    |
| `context`       | String | Extra contextual attributes added with the logger.                                               |

> **Note**: The Browser SDK will ignore modifications made to event properties not listed above. For more information about event properties, see the [Browser SDK repository][6].

Reviewer 3: Actually, since logs customers can already overwrite any property, we decided to allow the modifications of all properties with https://github.com/DataDog/browser-sdk/pull/909

> Allow modification of any field, not only sensitive fields.
> Allow addition of new fields.

Reviewer: ```suggestion
```

Reviewer: ^ Suggestion to remove the note following @bcaudan's input.

[2]: /account_management/api-app-keys/#client-tokens
[3]: https://www.npmjs.com/package/@datadog/browser-logs
[4]: https://github.com/DataDog/browser-sdk/blob/main/packages/logs/BROWSER_SUPPORT.md
[5]: /real_user_monitoring/guide/enrich-and-control-rum-data/

> [6]: https://github.com/DataDog/browser-sdk/blob/main/packages/logs/src/logsEvent.types.ts

Reviewer 2: This link isn't used anywhere I think?
```suggestion
```
